OpenRA.Game/Traits/TraitsInterfaces.cs

```diff
@@ -410,6 +410,27 @@ public interface IRenderAboveShroudWhenSelected
 		bool SpatiallyPartitionable { get; }
 	}
 
+	public interface ISelection
+	{
+		int Hash { get; }
+		IEnumerable<Actor> Actors { get; }
+
+		void SetNotifySelection(IEnumerable<INotifySelection> worldNotifySelection);
+		void Add(Actor a);
+		void Remove(Actor a);
+		void OnOwnerChanged(Actor a, Player oldOwner, Player newOwner);
```

Comment:
This can be removed from the interface by changing `Actor.ChangeOwnerSync` to query and call `INotifyOwnerChanged` on the world actor traits and changing `Selection` to implement that interface.

Reply:
> to query and call `INotifyOwnerChanged` on the world actor traits 

WorldActor doesn’t implement `INotifyOwnerChanged`. Some example may help